controllers/admin/admin_media_controller.py

```diff
@@ -26,6 +26,21 @@ def get(self):
         self.response.out.write(template.render(path, self.template_values))
 
 
+class AdminMediaDeleteReference(LoggedInHandler):
+    def post(self, media_key_name):
+        self._require_admin()
+
+        media = Media.get_by_id(media_key_name)
+
+        media.references.remove(media.create_reference(
+            self.request.get("reference_type"),
+            self.request.get("reference_key_name")))
+
+        media.put() ## Todo, update manipulators to support removing an item that ends a list
```

Comment:
I will add a parameter called `allow_deleting_data` to make it really explicit what you're doing when you set it :)